esmvalcore/_task.py

```diff
@@ -609,9 +609,10 @@ def _run(self, input_files):
 
         returncode = None
 
-        with resource_usage_logger(process.pid, self.resource_log), open(
-            self.log, "ab"
-        ) as log:
+        with (
+            resource_usage_logger(process.pid, self.resource_log),
+            open(self.log, "ab") as log,
```

Comment:
opening file as bytes object, but then adding to it? Not a negative remark, am just wondering why not as plaintext with encoding

Reply:
That saves decoding the text as it comes out of the process that runs the diagnostic script and then encoding it again to write it to file.